cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -382,16 +409,48 @@ func (sub CosmosSub) handleBurnLockMsg(cosmosMsg types.CosmosMsg, claimType type
 		"cosmosMsg", cosmosMsg,
 	)
 
-	err := txs.RelayProphecyClaimToEthereum(
-		sub.EthProvider,
-		sub.RegistryContractAddress,
-		claimType,
-		cosmosMsg,
-		sub.PrivateKey,
-		sub.SugaredLogger,
-	)
+	client, auth, target, err := tryInitRelayConfig(sub, claimType)
+	if err != nil {
+		sub.SugaredLogger.Errorw("failed in init relay config.",
+			errorMessageKey, err.Error())
+		return
+	}
 
+	// Initialize CosmosBridge instance
+	cosmosBridgeInstance, err := cosmosbridge.NewCosmosBridge(target, client)
 	if err != nil {
-		log.Println(err)
+		sub.SugaredLogger.Errorw("failed to get cosmosBridge instance.",
+			errorMessageKey, err.Error())
+		return
+	}
+
+	maxRetries := 5
+	i := 0
+
+	for i < maxRetries {
+		err = txs.RelayProphecyClaimToEthereum(
```

Comment:
would it make sense for this loop to be its own method, mirroring the tryInitRelayConfig method?

Reply:
Having the loop in this method makes the code clearer.